### ISSUE TITLE
The `Insert` API now only queries Groups internally at most once regardless of the quantity of RawContacts being inserted

### ISSUE THREAD

**Issue by vestrel00** [vestrel00/contacts-android#330]
## Problem

Internally, the `Insert` API queries the Groups table to get the Groups of the Account for every RawContact, 

> ℹ️ The following is pseudocode. 

```kotlin
for (rawContact in rawContacts) {
    val groupsList = getAccountGroupsList(rawContact.account)
    val validGroupMemberships  = getValidGroupMemberships(rawContact.groupMemberships, groupsList)
    ...
}
```

This slows down total insertion speed of multiple RawContacts, especially as the number of RawContacts being inserted increases.

## Solution

Query Groups once per insert operation (on `commit()`) instead,

> ℹ️ The following is pseudocode. 

```kotlin
val accountGroupsMap = getAccountGroupsMap(rawContacts.map { it.account })
for (rawContact in rawContacts) {
    val validGroupMemberships  = getValidGroupMemberships(rawContact.groupMemberships, accountGroupsMap[rawContact.account])
    ...
}
```

**Comment by vestrel00** [comment]
This was closed in https://github.com/vestrel00/contacts-android/commit/964ea16f8d225e4de97c122c91e7cc0a78a7c009#r128693894

**Comment by vestrel00** [comment]
This is included in release [0.3.2](https://github.com/vestrel00/contacts-android/releases/tag/0.3.2).
